### PR TITLE
Update gocd

### PR DESCRIPTION
**Pull request by hazim1093**
Updated gocd module to be able to use `ami-baker` from a GoCD job

### CODE REVIEW

File: modules/gocd/conf/cruise-config.xml [hunk 1]
```diff
@@ -122,7 +117,25 @@
               <exec command="/bin/bash" workingdir="#{SOURCE_DIR}">
                 <arg>-c</arg>
                 <arg>newTag=#{APP_DOCKER_IMAGE}:#{APP_IMAGE_BUILD_VERSION};\
-                     sudo docker push ${newTag}</arg>
+                     sudo docker push ${newTag};
+                     echo "Delete Empty Docker Images ...";
```
Comment (rasheedamir) [inline modules/gocd/conf/cruise-config.xml:121]: @hazim1093 wat do you mean by empty?

File: modules/gocd/conf/cruise-config.xml [hunk 2]
```diff
@@ -136,64 +149,36 @@
...
               <exec command="/bin/bash" workingdir="#{SOURCE_DIR}">
                 <arg>-c</arg>
-                <arg>cd #{STAKATER_LOCATION};sudo git pull origin master; sudo git fetch;sudo git checkout gocd-units;sudo git checkout -f; sudo git pull origin gocd-units;</arg>
+                <arg>cd /app/amibaker;sudo git pull origin master;</arg>
                 <runif status="passed" />
               </exec>
               <exec command="/bin/bash" workingdir="#{SOURCE_DIR}">
                 <arg>-c</arg>
-                <arg>sudo docker run -d --name terraform_${GO_PIPELINE_NAME} -v #{STAKATER_LOCATION}:/usr/src/app stakater/terraform</arg>
+                <arg>sudo docker run -d --name packer_${GO_PIPELINE_NAME} -v /app/amibaker:/usr/src/app stakater/packer</arg>
```
Comment (rasheedamir) [inline modules/gocd/conf/cruise-config.xml:163]: @hazim1093 can't we run packer natively? or we have to run it through docker ... just a thought... no need to do rite now; as we can do it later but should create a technical debt ticket

File: resources/policies/gocd_policy.json [hunk 3]
```diff
@@ -24,6 +24,37 @@
         "arn:aws:s3:::AWS-ACCOUNT-CLUSTER-NAME-config",
         "arn:aws:s3:::AWS-ACCOUNT-CLUSTER-NAME-config/*"
       ]
+    },
+    {
+      "Sid": "PackerPolicy",
```
Comment (rasheedamir) [inline resources/policies/gocd_policy.json:29]: @hazim1093 well done on this! seems like quite restrictive!

**Comment by rasheedamir** [comment]
@hazim1093 please delete the dead code which is related to older ami baker solution! and include it in this PR as well

**Comment by rasheedamir** [comment]
fixes #34 
fixes #37 
fixes #39

File: modules/gocd/conf/cruise-config.xml (hunk 1 above)
Comment (hazim1093) [inline modules/gocd/conf/cruise-config.xml:121]: when we tag new docker images as "latest" the older ones have empty tag and name. Refers to that.

File: modules/gocd/conf/cruise-config.xml (hunk 2 above)
Comment (hazim1093) [inline modules/gocd/conf/cruise-config.xml:163]: We can use a modified gocd-agent image which has packer installed

**Pull request merged**